cmd/opm/alpha/veneer/composite.go

```diff
@@ -0,0 +1,172 @@
+			setupFailed := false
+			setupErrors := map[string][]string{}
+			for _, catalog := range catalogConfig.Catalogs {
+				errs := []string{}
+				if catalog.Destination.BaseImage == "" {
+					errs = append(errs, "destination.baseImage must not be an empty string")
+				}
+
+				if catalog.Destination.WorkingDir == "" {
+					errs = append(errs, "destination.workingDir must not be an empty string")
+				}
+
+				// check for validation errors and skip builder creation if there are any errors
+				if len(errs) > 0 {
+					setupFailed = true
+					setupErrors[catalog.Name] = errs
+					continue
+				}
+
+				if _, ok := catalogBuilderMap[catalog.Name]; !ok {
+					builderMap := make(composite.BuilderMap)
+					for _, schema := range catalog.Builders {
+						builder, err := builderForSchema(schema, composite.BuilderConfig{
+							ContainerCfg: composite.ContainerConfig{
+								ContainerTool: containerTool,
+								BaseImage:     catalog.Destination.BaseImage,
+								WorkingDir:    catalog.Destination.WorkingDir,
+							},
+							OutputType:       output,
+							CurrentDirectory: wd,
+						})
+						if err != nil {
+							log.Fatalf("getting builder %q for catalog %q: %s", schema, catalog.Name, err)
+						}
+						builderMap[schema] = builder
+					}
+					catalogBuilderMap[catalog.Name] = builderMap
+				}
+			}
+
+			// if there were errors validating the catalog configuration then exit
+			if setupFailed {
+				//build the error message
+				var errMsg string
+				for cat, errs := range setupErrors {
+					errMsg += fmt.Sprintf("\nCatalog %s:\n", cat)
+					for _, err := range errs {
+						errMsg += fmt.Sprintf("  - %s\n", err)
+					}
+				}
+				log.Fatalf("catalog configuration file field validation failed: %s", errMsg)
+			}
+
+			veneer.CatalogBuilders = catalogBuilderMap
+
+			compositeData, err := os.Open(compositeFile)
+			if err != nil {
+				log.Fatalf("opening composite config file %q: %s", compositeFile, err)
+			}
+			defer compositeData.Close()
+
+			// parse data to composite config
+			compositeConfig := &composite.CompositeConfig{}
+			compositeDoc := json.RawMessage{}
+			compositeDecoder := yaml.NewYAMLOrJSONDecoder(compositeData, 4096)
+			err = compositeDecoder.Decode(&compositeDoc)
+			if err != nil {
+				log.Fatalf("decoding composite config: %s", err)
+			}
+			err = json.Unmarshal(compositeDoc, compositeConfig)
+			if err != nil {
+				log.Fatalf("unmarshalling composite config: %s", err)
+			}
+
+			if compositeConfig.Schema != composite.CompositeSchema {
+				log.Fatalf("%q has unknown schema, should be %q", compositeFile, composite.CompositeSchema)
+			}
+
+			err = veneer.Render(cmd.Context(), compositeConfig, validate)
+			if err != nil {
+				log.Fatalf("rendering the composite veneer: %s", err)
+			}
+		},
+	}
+	cmd.Flags().StringVarP(&output, "output", "o", "json", "Output format (json|yaml)")
+	// TODO: Investigate ways to do this without using a cli tool like docker/podman
```

Comment:
I was going to opine that we should allow use of (and with the same precedence ordering, default, etc.) as other `opm` sub-commands, but I think for this first pass I'd rather if we removed the option to specify (just comment out, don't delete) and always defaulted to docker.

Reply:
Fair enough, done in commit https://github.com/operator-framework/operator-registry/pull/1054/commits/bfde6b3074cc7e7a083048f085c2aac7e22c4c70

Reply:
i remember iib having issues with docker - can we check with them that omitting podman is ok?

Reply:
The ideal thing to do here will be to have the containerd implementation as the underlying component, that way we don't have dependency on either docker/podman/etc. 

But as long as it's called out here as the "must do before graduating from alpha" it doesn't need to block this particular PR.